share/completions/istioctl.fish

```diff
@@ -0,0 +1,176 @@
+
+    __istioctl_debug "Calling $requestComp"
+    set -l results (eval $requestComp 2> /dev/null)
+
+    # Some programs may output extra empty lines after the directive.
+    # Let's ignore them or else it will break completion.
+    # Ref: https://github.com/spf13/cobra/issues/1279
+    for line in $results[-1..1]
+        if test (string trim -- $line) = ""
+            # Found an empty line, remove it
+            set results $results[1..-2]
+        else
+            # Found non-empty line, we have our proper output
+            break
+        end
+    end
+
+    set -l comps $results[1..-2]
+    set -l directiveLine $results[-1]
+
+    # For Fish, when completing a flag with an = (e.g., <program> -n=<TAB>)
+    # completions must be prefixed with the flag
+    set -l flagPrefix (string match -r -- '-.*=' "$lastArg")
+
+    __istioctl_debug "Comps: $comps"
+    __istioctl_debug "DirectiveLine: $directiveLine"
+    __istioctl_debug "flagPrefix: $flagPrefix"
+
+    for comp in $comps
+        printf "%s%s\n" "$flagPrefix" "$comp"
+    end
+
+    printf "%s\n" "$directiveLine"
+end
+
+# This function does two things:
+# - Obtain the completions and store them in the global __istioctl_comp_results
+# - Return false if file completion should be performed
+function __istioctl_prepare_completions
+    __istioctl_debug ""
+    __istioctl_debug "========= starting completion logic =========="
+
+    # Start fresh
+    set --erase __istioctl_comp_results
+
+    set -l results (__istioctl_perform_completion)
+    __istioctl_debug "Completion results: $results"
+
+    if test -z "$results"
+        __istioctl_debug "No completion, probably due to a failure"
+        # Might as well do file completion, in case it helps
+        return 1
+    end
+
+    set -l directive (string sub --start 2 $results[-1])
+    set --global __istioctl_comp_results $results[1..-2]
+
+    __istioctl_debug "Completions are: $__istioctl_comp_results"
+    __istioctl_debug "Directive is: $directive"
+
+    set -l shellCompDirectiveError 1
+    set -l shellCompDirectiveNoSpace 2
+    set -l shellCompDirectiveNoFileComp 4
+    set -l shellCompDirectiveFilterFileExt 8
+    set -l shellCompDirectiveFilterDirs 16
+
+    if test -z "$directive"
+        set directive 0
+    end
+
+    set -l compErr (math (math --scale 0 $directive / $shellCompDirectiveError) % 2)
+    if test $compErr -eq 1
+        __istioctl_debug "Received error directive: aborting."
+        # Might as well do file completion, in case it helps
+        return 1
+    end
+
+    set -l filefilter (math (math --scale 0 $directive / $shellCompDirectiveFilterFileExt) % 2)
+    set -l dirfilter (math (math --scale 0 $directive / $shellCompDirectiveFilterDirs) % 2)
+    if test $filefilter -eq 1; or test $dirfilter -eq 1
+        __istioctl_debug "File extension filtering or directory filtering not supported"
+        # Do full file completion instead
```

Comment:
We no longer do this. Either the completions say files are okay or they don't.

There are very few options that allow files here - basically the configuration file options (according to https://istio.io/latest/docs/reference/commands/istioctl/). They can simply be added directly, with

```fish
complete -c istioctl -l kubeconfig -rF
```

Reply:
done